src/server/scripts/Northrend/zone_zuldrak.cpp

```diff
@@ -1196,6 +1201,105 @@ class spell_zuldrak_gymers_throw : public SpellScript
     }
 };
 
+//Quest 12686 Zero Tolerance
+
+enum
+{
+    SPELL_TRIGGER_ID = 52390,
+    NPC_ENTRY_CONTROLLED = 28805,
+    NPC_ENTRY_NORMAL = 28802,
+    QUEST_ID = 12686,
+    SPELL_CAST = 50361,
+
+    EVENT_CAST_SPELL = 1,
+    EVENT_CONTROL_CHECK = 2,
+};
+
+class npc_servant_of_drakuru : public CreatureScript
+{
+public:
+    npc_servant_of_drakuru() : CreatureScript("npc_servant_of_drakuru") {}
+
+    struct npc_servant_of_drakuruAI : public ScriptedAI
+    {
+        npc_servant_of_drakuruAI(Creature* creature): ScriptedAI(creature), _controller(nullptr), _controlled(false) { }
+
+        void JustEngagedWith(Unit* /*who*/) override
+        {
+            _events.Reset();
+            _events.ScheduleEvent(EVENT_CAST_SPELL, 4s);
+            _events.ScheduleEvent(EVENT_CONTROL_CHECK, 1s);
+        }
+
+        void SpellHit(WorldObject* caster, SpellInfo const* spell) override
+        {
+            if (spell->Id == SPELL_TRIGGER_ID)
+            {
+                if (Player* player = caster->ToPlayer())
+                {
+                    if (player->GetQuestStatus(QUEST_ID) == QUEST_STATUS_INCOMPLETE)
+                    {
+                        _controller = player;
+                        Talk(0);
+                        me->UpdateEntry(NPC_ENTRY_CONTROLLED);
+                        me->SetHealth(me->GetMaxHealth());
+                        _controlled = true;
+                    }
+                }
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_CAST_SPELL:
+                    if (!_controlled)
+                    {
+                        if (Unit* target = me->GetVictim())
+                            me->CastSpell(target, SPELL_CAST, true);
+                    }
+                    _events.ScheduleEvent(EVENT_CAST_SPELL, 12s, 14s);
+                    break;
+
+                case EVENT_CONTROL_CHECK:
+                    if (_controlled)
+                    {
+                        if (me->GetCharmerOrOwner() != _controller)
+                        {
+                            if (_controller && _controller->GetQuestStatus(QUEST_ID) == QUEST_STATUS_INCOMPLETE)
+                                me->DespawnOrUnsummon(1s, 3s);
```

Comment:
You're using forceRespawnTimer but creature can't respawn (and shouldn't)
It should be just 
```cpp
me->DespawnOrUnsummon(1s);
```